android/build.gradle

```diff
@@ -4,14 +4,41 @@ apply from: './jacoco.gradle'
 apply from: './native.gradle'
 apply from: './sourcemaps.gradle'
 
+/* Checking APG version to be backward-compatible with RN versions < 0.71 */
+def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')[0].toInteger()
+def shouldUseNameSpace = agpVersion >= 7
+def PACKAGE_PROP = "package=\"com.instabug.reactlibrary\""
+def manifestOutFile = file("${projectDir}/src/main/AndroidManifest.xml")
+def manifestContent = manifestOutFile.getText()
+
 String getExtOrDefault(String name) {
     if (rootProject.ext.has(name)) {
         return rootProject.ext.get(name)
     }
     return project.properties['InstabugReactNative_' + name]
 }
 
+if(shouldUseNameSpace){
+      manifestContent = manifestContent.replaceAll(
+        PACKAGE_PROP,
+        ''
+    )  
+} else {
+    if(!manifestContent.contains("$PACKAGE_PROP")){
+        manifestContent = manifestContent.replace(
+            '<manifest',
+            "<manifest $PACKAGE_PROP "
+        )
+    }
+}
+
+manifestContent.replaceAll("  ", " ")
+manifestOutFile.write(manifestContent)
```

Comment:
a suggestion would be to write to manifest only if we made a change to it. We can check for change then write to the manifestFile.

```
if (manifestOutFile.getText() != manifestContent) {
    manifestOutFile.write(manifestContent)
}
```

We can also add a new property `updatedManifestContent` for better readability and use it in checking for changes before writing to the `manifestOutFile`.

Reply:
Yes, I agree with that
Will resolve this and push a new update.